src/coprocessor/dag/executor/scanner.rs

```diff
@@ -150,32 +154,24 @@ impl<S: Snapshot> Scanner<S> {
         if self.no_more {
             return false;
         }
-        if self.last_scanned_key.is_empty() {
-            // No more future requests will be send, so this is acceptable.
-            match self.scan_mode {
-                ScanMode::Forward => range.set_end(self.range.get_start().to_owned()),
-                ScanMode::Backward => range.set_start(self.range.get_end().to_owned()),
-                _ => unreachable!(),
+        // `next_row()` should have been called when calling `stop_scan()` after `start_scan()`.
+        assert!(!self.last_scanned_key.is_empty());
+        match self.scan_mode {
+            ScanMode::Forward => {
+                // Make range_end exclusive. Note that next `start_scan` will also use this
+                // key as the range_start.
+                util::convert_to_prefix_next(&mut self.last_scanned_key);
+                range.set_end(self.last_scanned_key.clone());
             }
-        } else {
-            match self.scan_mode {
-                ScanMode::Forward => {
-                    // Make range_end exclusive. Note that next `start_scan` will also use this
-                    // key as the range_start.
-                    util::convert_to_prefix_next(&mut self.last_scanned_key);
-                    range.set_end(self.last_scanned_key.clone());
+            ScanMode::Backward => {
+                // TODO: We may don't need `truncate_as_row_key`. Needs investigation.
+                if self.scan_on == ScanOn::Table {
+                    let row_key_len = truncate_as_row_key(&self.last_scanned_key).unwrap().len();
```

Comment:
Should remove `truncate_as_row_key`.

Reply:
Let's leave the TODO here and remove it after investigation, since according to @Connor1996 
 we are not sure that everywhere we passed a key that ts was truncated.